dev/release/verify-yum.sh

```diff
@@ -99,13 +110,15 @@ case "${distribution}-${distribution_version}" in
     uninstall_command="yum remove -y"
     clean_command="yum clean"
     info_command="yum info"
+    should_fix_eol_repositories=yes
```

Comment:
Could you run `fix_eol_repositories` here because this is needed not only `${TYPE} = local` but also all `${TYPE}`?

Reply:
Something like https://github.com/apache/arrow/pull/42129/commits/504a36a49c71536d0163ce7348d8eb8f7b302516?

Reply:
I tested the change locally against centos 7 and stream-8 and it still works the same so that's good.

Reply:
Resolved by https://github.com/apache/arrow/pull/42129/commits/504a36a49c71536d0163ce7348d8eb8f7b302516.